reconcile/ocm_clusters.py

```diff
@@ -83,153 +255,73 @@ def run(dry_run, gitlab_project_id=None, thread_pool_size=10):
-            if desired_provision_shard_id != current_provision_shard_id:
-                clusters_updates[cluster_name]["spec"][
-                    "provision_shard_id"
-                ] = current_provision_shard_id
-
-            if clusters_updates[cluster_name]:
-                clusters_updates[cluster_name]["path"] = cluster_path
-
-            # exclude params we don't want to check in the specs
-            for k in ["id", "external_id", "provision_shard_id"]:
-                current_spec["spec"].pop(k, None)
-                desired_spec["spec"].pop(k, None)
-
-            desired_uwm = desired_spec["spec"].get(ocmmod.DISABLE_UWM_ATTR)
-            current_uwm = current_spec["spec"].get(ocmmod.DISABLE_UWM_ATTR)
-
-            if desired_uwm is None and current_uwm is not None:
-                clusters_updates[cluster_name]["spec"][
-                    ocmmod.DISABLE_UWM_ATTR
-                ] = current_uwm  # noqa: E501
-
-            # check if cluster update, if any, is valid
-            update_spec, err = get_cluster_update_spec(
-                cluster_name,
-                current_spec,
-                desired_spec,
+            # OCM API Updates
+            # Changes made to pp-interface manifests that need to be requested
+            # to the OCM Api
+            ocm = ocm_map.get(cluster_name)
+            update_spec, err = get_cluster_ocm_update_spec(
+                ocm, cluster_name, current_spec, desired_spec
             )
             if err:
-                logging.warning(f"Invalid changes to spec: {update_spec}")
                 error = True
                 continue
+
             # update cluster
             # TODO(mafriedm): check dry_run in OCM API patch
             if update_spec:
                 logging.info(["update_cluster", cluster_name])
                 logging.debug(
-                    "[%s] desired spec %s is different " + "from current spec %s",
-                    cluster_name,
-                    desired_spec,
-                    current_spec,
+                    f"current_spec: {current_spec}, desired_spec: {desired_spec}"
                 )
                 if not dry_run:
                     ocm = ocm_map.get(cluster_name)
-                    ocm.update_cluster(cluster_name, update_spec, dry_run)
+                    try:
+                        ocm.update_cluster(cluster_name, update_spec, dry_run)
+                    except NotImplementedError:
+                        logging.error(
+                            f"[{cluster_name}] Update clusters is currently not "
+                            "implemented for [{desired_spec.spec.product}] product. "
+                            "Updates to the cluster spec are not supported."
+                        )
+                        # Not marking error as a changer made in ocm could trigger
+                        # this.
+                        # error = True
```

Comment:
can you elaborate on this?

Reply:
I've improved the comment. I want to cover the case where a product does not implement cluster updates. If a change is made in OCM out of app interface, the integration will log the error but the integration will not fail. This could be useful in cases where we want to have the cluster reference but not manage it through ocm-clusters integration.